common/commit_interrupt.go

```diff
@@ -0,0 +1,9 @@
+package common
+
+import "sync/atomic"
+
+// InterruptBlockBuilding is a global flag which will be set to true to notify
+// the miner to stop building the current block. It's also used in EVM interpreter
+// to interrupt an ongoing transaction execution. It's reset to false when block
+// building time of the next block starts.
+var InterruptBlockBuilding atomic.Bool
```

Comment:
It seems a bit dangerous to use a global variable. Why can't this be created per mining task?

Reply:
It seems to me that the approach without using global variable is to directly change the interrupt context type [here](https://github.com/maticnetwork/bor/blob/0cebe7369811af60bb5cc01724f79d154f201a7c/miner/worker.go#L270). This would involve some type changes but won't be too much I guess? 
If going with global variable approach, the interruptCtx would basically become unless right?

Reply:
I don't think we'll be able to completely remove the `interruptCtx` as we still need to check for `Done()` in the EVM interpreter but let me see if there's some other workaround for that.

Reply:
Refactored a bunch of this which also handles this concern. Thanks!